src/controllers/playback/video/index.js

```diff
@@ -326,7 +326,93 @@ export default function (view) {
         elem.removeEventListener(transitionEndEventName, onHideAnimationComplete);
     }
 
-    const _focus = debounce((focusElement) => focusManager.focus(focusElement), 50);
+    /*const _focus = function () {
+        let lastFocusElement;
+        let timerId;
+
+        function invokeFunc() {
+            // If no focus element is provided, try to keep current focus if it's valid,
+            // otherwise default to pause button
+            const currentFocus = lastFocusElement || document.activeElement;
+            if (!currentFocus || !focusManager.isCurrentlyFocusable(currentFocus)) {
+                lastFocusElement = osdBottomElement.querySelector('.btnPause');
+            }
+
+            if (lastFocusElement) focusManager.focus(lastFocusElement);
+
+            lastFocusElement = undefined;
+            timerId = undefined;
+        }
+
+        const debounced = function (focusElement) {
+            if (focusElement) lastFocusElement = focusElement;
+
+            if (timerId === undefined) {
+                timerId = setTimeout(invokeFunc, 50);
+            }
+        };
+
+        function cancel() {
+            if (timerId !== undefined) {
+                clearTimeout(timerId);
+            }
+            lastFocusElement = undefined;
+            timerId = undefined;
+        }
+
+        debounced.cancel = cancel;
+
+        return debounced;
+    }();*/
+    const _focus = function () {
+        let lastFocusElement;
+
+        const debouncedInternal = debounce(() => {
+            // If no focus element is provided, try to keep current focus if it's valid,
+            // otherwise default to pause button
+            const currentFocus = lastFocusElement || document.activeElement;
+            if (!currentFocus || !focusManager.isCurrentlyFocusable(currentFocus)) {
+                lastFocusElement = osdBottomElement.querySelector('.btnPause');
+            }
+
+            if (lastFocusElement) focusManager.focus(lastFocusElement);
+
+            lastFocusElement = undefined;
+        }, 50);
+
+        const debounced = function (focusElement) {
+            if (focusElement) lastFocusElement = focusElement;
+
+            debouncedInternal();
+        };
+
+        function cancel() {
+            debouncedInternal.cancel();
+            lastFocusElement = undefined;
+        }
+
+        function flush() {
+            const result = debouncedInternal.flush();
+            lastFocusElement = undefined;
+            return result;
+        }
+
+        debounced.cancel = cancel;
+        debounced.flush = flush;
+
+        return debounced;
+    }();
+
+    const _focus2 = function (focusElement) {
+        // If no focus element is provided, try to keep current focus if it's valid,
+        // otherwise default to pause button
+        const currentFocus = focusElement || document.activeElement;
```

Comment:
So the main fix here is always making sure this check runs right?

What does wrapping the skip button in a container do?

Reply:
> So the main fix here is always making sure this check runs right?

More precisely, `currentFocus` should be a valid element for `focusManager.isCurrentlyFocusable(currentFocus)`.

> What does wrapping the skip button in a container do?

`FocusManager.isCurrentlyFocusableInternal` doesn't work with
fixed elements - they aren't considered focusable.